shell/platform/fuchsia/dart_runner/dart_test_component_controller_v2.h

```diff
@@ -0,0 +1,157 @@
+  /// false otherwise.
+  bool RunDartMain();
+
+  /// Creates and binds the namespace for this component. Returns true if
+  /// successful, false otherwise.
+  bool CreateAndBindNamespace();
+
+  bool SetUpFromKernel();
+  bool SetUpFromAppSnapshot();
+
+  bool CreateIsolate(const uint8_t* isolate_snapshot_data,
+                     const uint8_t* isolate_snapshot_instructions);
+
+  // |ComponentController|
+  void Kill() override;
+  void Stop() override;
+
+  // Idle notification.
+  void MessageEpilogue(Dart_Handle result);
+  void OnIdleTimer(async_dispatcher_t* dispatcher,
+                   async::WaitBase* wait,
+                   zx_status_t status,
+                   const zx_packet_signal* signal);
+
+  // |CaseIterator|
+  class CaseIterator final : public fuchsia::test::CaseIterator {
+   public:
+    CaseIterator(fidl::InterfaceRequest<fuchsia::test::CaseIterator> request,
+                 async_dispatcher_t* dispatcher,
+                 fit::function<void(CaseIterator*)> done_callback);
+
+    void GetNext(GetNextCallback callback) override;
+
+   private:
+    int get_next_call_count = 0;
+    fidl::Binding<fuchsia::test::CaseIterator> binding_;
+    fit::function<void(CaseIterator*)> done_callback_;
+  };
+
+  std::unique_ptr<CaseIterator> RemoveCaseInterator(CaseIterator*);
+
+  // We only need one case_listener currently as dart tests are run as one
+  // large test file. In future iterations, case_listeners must be
+  // created per test case.
+  fidl::InterfacePtr<fuchsia::test::CaseListener> case_listener_;
+  std::map<CaseIterator*, std::unique_ptr<CaseIterator>> case_iterators_;
+
+  // |Suite|
+
+  /// Exposes suite protocol on behalf of test component.
+  std::unique_ptr<sys::ComponentContext> suite_context_;
+  fidl::BindingSet<fuchsia::test::Suite> suite_bindings_;
+
+  // The loop must be the first declared member so that it gets destroyed after
+  // binding_ which expects the existence of a loop.
+  std::unique_ptr<async::Loop> loop_;
+
+  std::string label_;
+  std::string url_;
+  std::shared_ptr<sys::ServiceDirectory> runner_incoming_services_;
+  std::string data_path_;
+  std::unique_ptr<sys::ComponentContext> context_;
+
+  fuchsia::component::runner::ComponentStartInfo start_info_;
+  fidl::Binding<fuchsia::component::runner::ComponentController> binding_;
+  DoneCallback done_callback_;
+
+  fdio_ns_t* namespace_ = nullptr;
+  int stdout_fd_ = -1;
+  int stderr_fd_ = -1;
+
+  dart_utils::ElfSnapshot elf_snapshot_;                      // AOT snapshot
+  dart_utils::MappedResource isolate_snapshot_data_;          // JIT snapshot
+  dart_utils::MappedResource isolate_snapshot_instructions_;  // JIT snapshot
+  std::vector<dart_utils::MappedResource> kernel_peices_;
+
+  Dart_Isolate isolate_;
+  int32_t return_code_ = 0;
+
+  zx::time idle_start_{0};
+  zx::timer idle_timer_;
+  async::WaitMethod<DartTestComponentControllerV2,
```

Comment:
We got rid of the logic that starts idle_wait_ https://photos.app.goo.gl/Q3zEMzuDHiCo56vz5 so are we sure this and OnIdleTimer() is doing something? or can they be removed?

Reply:
I'll bring back the logic. Just want to maintain similar execution between this component and the other V2 component implementations.